Networking/Networking/ApplicationPassword/ApplicationPasswordUseCase.swift

```diff
@@ -62,29 +63,37 @@ final class DefaultApplicationPasswordUseCase: ApplicationPasswordUseCase {
         }
     }
 
-    init(siteID: Int64,
-         networkcredentials: Credentials,
-         network: Network? = nil,
-         keychain: Keychain = Keychain(service: KeychainServiceName.name)) {
-        self.siteID = siteID
-        self.credentials = networkcredentials
-        self.keychain = keychain
+    @MainActor
+    init(username: String,
+         password: String,
+         siteAddress: String,
+         network: Network? = nil) async throws {
+        self.siteAddress = siteAddress
+        self.username = username
 
         if let network {
             self.network = network
         } else {
-            self.network = ApplicationPasswordNetwork(credentials: networkcredentials)
+            guard let loginURL = URL(string: siteAddress + Constants.loginPath),
+                  let adminURL = URL(string: siteAddress + Constants.adminPath) else {
+                DDLogWarn("⚠️ Cannot construct login URL and admin URL for site \(siteAddress)")
+                throw ApplicationPasswordUseCaseError.invalidSiteAddress
+            }
+            // Prepares the authenticator with username and password
+            let authenticator = CookieNonceAuthenticator(username: username,
+                                                         password: password,
+                                                         loginURL: loginURL,
+                                                         adminURL: adminURL,
+                                                         version: Constants.defaultWPVersion,
+                                                         nonce: nil)
+            self.network = ApplicationPasswordNetwork(authenticator: authenticator)
```

Comment:
Can we reuse `WordPressOrgNetwork` instead of creating `ApplicationPasswordNetwork`? Previously we created `ApplicationPasswordNetwork` instead of using `AlamofireNetwork` to avoid circular dependency. This should no longer be a problem now.

If necessary, we can maybe rename `WordPressOrgNetwork` to `CookieNonceNetwork` to clarify the authentication method for the network.

Reply:
Good idea. Done in https://github.com/woocommerce/woocommerce-ios/pull/8457/commits/1eb988657e54059624e8cc9975ee9cf81064ddee

I think a rename isn't needed. `WordPressOrgNetwork` is clear IMO.